Sources/MapboxNavigation/UserLocationStyle.swift

```diff
@@ -0,0 +1,27 @@
+import Foundation
+import MapboxMaps
+
+/**
+ A type that represents a `UIView` that is `CourseUpdatable`.
+ */
+public typealias UserCourseView = UIView & CourseUpdatable
+
+/**
+ The style of the user location indicator in a `NavigationMapView`.
+ */
+public enum UserLocationStyle {
+    /**
+     The course view representing the user’s location and course on the map, switching between `UserPuckCourseView` and `UserHaloCourseView` based on the level of location accuracy.
+     */
+    case courseView(configuration: UserCourseView)
```

Comment:
The associated value is the view to display, not a configuration object. Since the enumeration name already says “view”, there’s no need for a separate label:

```swift
case courseView(_ view: UserCourseView)
```

Reply:
Thanks for the suggestion, I'll update this part.